src/bools.jl

```diff
@@ -40,7 +40,7 @@
                 end
             end
         else
-            intx, intcode, intpos = typeparser(UInt8, source, pos, len, b, code, options)
+            intx, intcode, intpos = typeparser(conf(UInt8, options), source, pos, len, b, code, options)
```

Comment:
hmmmm.....how does this work exactly? the user can still pass `Options` and in this branch we're going to create an `IntConf` from `Options`?

Reply:
are we worried about extra allocations/performance here due to extra conversion/construction steps?

Reply:
Admittedly, I haven't really profiled this implementation, but I was expecting that the overhead would be pretty minimal since all `T -> Conf{T}` transform should be pretty inferrable and we shouldn't be allocating anything on the heap. Do we have some comprehensive benchmarks we could run to evaluate the impact here?